zerver/tests/test_events.py

```diff
@@ -4558,6 +4558,40 @@ def do_test_subscribe_events(self, include_subscribers: bool) -> None:
             ),
         )
 
+        moderators_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MODERATORS,
+            is_system_group=True,
+            realm=self.user_profile.realm,
+        )
+        with self.verify_action(include_subscribers=include_subscribers, num_events=1) as events:
+            do_change_stream_group_based_setting(
+                stream,
+                "can_administer_channel_group",
+                moderators_group,
+                acting_user=self.example_user("hamlet"),
```

Comment:
This feels artificial; `hamlet` doesn't have permission to make that edit, right?

Seems like a bug across this test. This is one reason we prefer to use the API rather than `do_` functions for most tests; you can write impossible scenarios this way. But I guess `test_events` is usually using the smaller `do_` functions rather than the API itself, so not isolated to what we're doing here.